xptifw/samples/sycl_perf_collector/sycl_perf_collector.cpp

```diff
@@ -0,0 +1,891 @@
+
+    char *Name;
+    // Register string - it may have already been registered, so we will get a
+    // populated string pointer that has a lifetime of XPTI framework
+    xptiRegisterString((UserData ? (const char *)UserData : "unknown"), &Name);
+    xpti::record_t r;
+
+    if (TraceType & 0x0001) {
+      // We are the closing scope step
+      {
+        std::lock_guard<std::mutex> _{GRecMutex};
+        auto ele = GRecordsInProgress->find(Instance);
+        if (ele != GRecordsInProgress->end()) {
+          // Copy so the incomplete record can be deleted
+          r = ele->second;
+          GRecordsInProgress->erase(ele);
+        } else {
+          throw std::runtime_error("Instance id/correlation ID collision!");
+        }
+        // We are operating on a copy, so no data races
+        record_state(r, false);
+        if (ShowDebugInformation)
+          print_record(TraceType, r);
+        // if (GDataModel.get())
+        if (GDataModel)
+          GDataModel->add(r);
+      }
+    } else {
+      // Create record and save as we are at the begin scope step
+      record_state(r, true);
+      r.Name = Name;
+      if (GOutputFormats & (uint64_t)xpti::FileFormat::JSON)
+        r.Category = StreamName;
+      r.Flags |= (uint64_t)(xpti::RecordFlags::NamePresent);
+      {
+        std::lock_guard<std::mutex> _{GRecMutex};
+        r.CorrID = Instance;
+        if (GRecordsInProgress->count(Instance)) {
+          throw std::runtime_error("Instance id/correlation ID collision!");
+        }
+        (*GRecordsInProgress)[Instance] = r;
+      }
+    }
+  }
+}
+
+XPTI_CALLBACK_API void traceCallback(uint16_t TraceType,
+                                     xpti::trace_event_data_t *Parent,
+                                     xpti::trace_event_data_t *Event,
+                                     uint64_t Instance, const void *UserData) {
+  if (CalibrationRun)
+    return;
+
+  const char *UD = (Event->reserved.payload->name)
+                       ? Event->reserved.payload->name
+                       : "Unknown";
+  record_and_save("sycl", (Event ? Event : Parent), TraceType, Instance, UD);
+}
+
+XPTI_CALLBACK_API
+void graphCallback(uint16_t TraceType, xpti::trace_event_data_t *Parent,
+                   xpti::trace_event_data_t *Event, uint64_t Instance,
+                   const void *UserData) {
+  if (CalibrationRun)
+    return;
+  // Need to add DOT writer here
+}
+
+XPTI_CALLBACK_API void syclMemCallback(uint16_t TraceType,
+                                       xpti::trace_event_data_t *Parent,
+                                       xpti::trace_event_data_t *Event,
+                                       uint64_t Instance,
+                                       const void *UserData) {
+  if (CalibrationRun)
+    return;
+  record_and_save("sycl.experimental.mem_alloc", (Event ? Event : Parent),
+                  TraceType, Instance,
+                  (TraceType & 0x0001) ? "Memory_allocation_end"
```

Comment:
why capital latters? would be better to use the same style as for other traces.

Reply:
Agreed, was probably a typo.